relay-server/src/service.rs

```diff
@@ -87,6 +90,7 @@ pub fn create_runtime(name: &str, threads: usize) -> Runtime {
 pub struct ServiceState {
     config: Arc<Config>,
```

Comment:
`ServiceState` is cloned very often by axum, and also from us. Cloning it is somewhat expensive now, since we clone a lot of fields: All the runtimes, and then all fields in the registry. It would now make sense to change the internals of `ServiceState` to something like:


```rust
#[derive(Clone, Debug)]
struct ServiceState {
    inner: Arc<StateInner>,
}

impl ServiceState {
    // accessor methods for every service here, accessing through `self.inner.*`
}

#[derive(Debug)] // no Clone!
struct StateInner {
    // all Addrs here
    // all runtimes here
    // buffer guard
}
```